### PR TITLE
fix problem with invalid time if no rtc

### PR DESCRIPTION
**Pull request by giggsoff**
I noticed, that I cannot onboard RPi with Adam and the same problem for defined rtc `base = "1970-01-01"` in QEMU.
The problem, was in certificate time mismatch with local inside EVE.

So, we need to remove `SenderStatusCertInvalid` because it is not `remoteTemporaryFailure` and depends on ntpd response.

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>

### CODE REVIEW

**Review by eriknordmark — CHANGES_REQUESTED** [review]
The SenderStatusCertInvalid case is when the controller has let its cert expire. In that case EVE patiently waits. Don't want it to conclude that the DevicePortConfig is bad and try other DevicePortConfig in that case.

Do you have a raspberry pi which thinks it is 1970?
Or does Adam/Eden think it is 1970?

We have a hack in device-steps.sh for the former.
But  we've also seen some hardwhich which thought it was Jan 1st some other year.
In such a case if we can't get an IP address plus NTP to work after 5 minutes (waitforaddr wait for 5) then the device might not know what time it is when it generates the device certificate.

Short of waiting forever for NTP (and what if the NTP protocol is blocked by a firewall?) we have a limited wait and basically assume that the hardware has a somewhat sane real-time clock.

Please explain what problem you are trying to solve.

**Comment by giggsoff** [comment]
Yes, RPi4 have no RTC module, so it think, that it is 1970. And yes, ntp do its work fine, time change on my RPi during device-steps ntp starting.
My QEMU with EVE also can think, that it is 1970. I do it manually for testing via:
```
[rtc]
  base = "1970-01-01"
  clock = "vm"
```

And in both variants I cannot see any attempts of calling `/register` endpoint of Adam.
I see comment about it here: https://github.com/lf-edge/eve/blob/788d85fa879806f6f4a36b127508c61711abee32/pkg/pillar/cmd/nim/nim.go#L907-L911
We are forced to launch [nim](https://github.com/lf-edge/eve/blob/788d85fa879806f6f4a36b127508c61711abee32/pkg/pillar/scripts/device-steps.sh#L289) before [ntp](https://github.com/lf-edge/eve/blob/788d85fa879806f6f4a36b127508c61711abee32/pkg/pillar/scripts/device-steps.sh#L311). And we give up upon receiving an answer `SenderStatusCertInvalid`.

**Comment by giggsoff** [comment]
I added a workaround to device-steps.sh to set the date based on the last modified time of root-certificate.pem.

**Comment by eriknordmark** [comment]
> We are forced to launch [nim](https://github.com/lf-edge/eve/blob/788d85fa879806f6f4a36b127508c61711abee32/pkg/pillar/scripts/device-steps.sh#L289) before [ntp](https://github.com/lf-edge/eve/blob/788d85fa879806f6f4a36b127508c61711abee32/pkg/pillar/scripts/device-steps.sh#L311). And we give up upon receiving an answer `SenderStatusCertInvalid`.

But thedevice cert isn't created until we have gotten out of 1970.
Are you seeing an error due to the server cert not being valid (NotBefore check)?

The root certificate might still be issued in 2018 and the server cert have a NotBefore of 2020, thus the fix doesn't work in all cases. 

I think a better approach would be to have VerifyDevicePortConfig() treat DPC_REMOTE_WAIT as DPC_SUCCESS (i.e., log as error, but proceed as if it was a success because we did manage to connect to the controller). This assumes that we don't have a bad https proxy which causes the CertInvalid error - in such a case we'd want to try a network path which does not have that bad proxy.)

**Comment by giggsoff** [comment]
Thanks for pointing out the direction of the modifications! Made edits.

**Comment by rvs** [comment]
Hey @giggsoff -- why don't you implement the simplest workaround of all: if the current time is less that the timestamp on device-steps.sh itself -- adjust the current time to be that of device-steps.sh.

Everything else on the system can be screwed up -- that date (or any other file in the immutable portion of EVE) is guaranteed not to be screwed up.

This is simple and will get us going on RPi4.

**Comment by giggsoff** [comment]
If we are going to use the device-steps.sh timestamp as the initial timestamp (https://github.com/itmo-eve/eve/commit/d3b9df0c9dc1d5be031a35b06cbd76ff3e056580), we need to change the Eden logic to issue certificates from NotBefore in the past (before the moment of building the pillar of EVE).

File: pkg/pillar/devicenetwork/dnc.go [hunk 1]
```diff
@@ -415,7 +415,9 @@ func VerifyDevicePortConfig(ctx *DeviceNetworkContext) {
 			SetupVerify(ctx, nextIndex)
 			continue
 
-		case types.DPC_SUCCESS:
+		case types.DPC_SUCCESS, types.DPC_REMOTE_WAIT:
```
Comment (eriknordmark) [inline pkg/pillar/devicenetwork/dnc.go:418]: Please fix the logs below to change the hard-coded DPC_SUCCESS to res.String()

**Review by eriknordmark — COMMENTED** [review]
One nit about logging output, then we'll pull

**Review by eriknordmark — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**